core/CliMulti/Process.php

```diff
@@ -81,6 +82,10 @@ public function hasStarted($content = null)
 
     public function hasFinished()
     {
+        if ($this->finished) {
```

Comment:
technically, I was thinking we should check also if the process `hasStarted()` and when `not has started` not declare it as finished. It might be good to eventually do that especially once https://github.com/matomo-org/matomo/pull/16950/files is merged and we then remember if it ever started. The only problem be if a process starts and finishes too quick as part of climulti and then we might not detect the start and therefore don't declare it as finished and then climulti would wait for 8s to declare it as finished but this could happen anyway I suppose https://github.com/matomo-org/matomo/blob/4.0.5/core/CliMulti.php#L217-L222. 

I might add this in a separate PR.

Reply:
To get around the process quickly starts & finishes issue, we could avoid deleting pid files, and maybe delete them in a task. We'd have to include a timestamp somewhere, though..

I'm not sure how much of a problem it is re archiving though, since those jobs should take a bit to start.

Reply:
Yeah I think it's not really an issue and it be rare. Currently, the deletion of the pid file indicates a finished process but we can see. It shouldn't be needed for now I think (I think the method isn't even really called or so unless the process has started etc). I'll see if our issues are fixed with all these changes applied and if not investigate further